warehouse/accounts/forms.py

```diff
@@ -286,7 +286,7 @@ def validate_email(self, field):
             )
 
         # Check if the domain is valid
-        domain = field.data.split("@")[-1]
+        domain = ".".join(field.data.split("@")[-1].split(".")[-2:]).lower()
```

Comment:
I think this is fine, but I guess I don't understand why we aren't re-using the `Address` we create on L282 above. eg:

```python
address = Address(addr_spec="foo@clan.wutang.net")
assert address.domain == "clan.wutang.net"

domain = ".".join(address.domain.split(".")[-2:]).lower()
assert domain == 'wutang.net'
```

It doesn't save a ton, but tis line conditional is getting more complex to read, and we're already doing the work to parse it.

Dealer's choice!